sklearn/tests/test_pipeline.py

```diff
@@ -557,13 +557,17 @@ def test_pipeline_fit_transform():
 
 
 def test_pipeline_slice():
-    pipe = Pipeline([('transf1', Transf()),
-                     ('transf2', Transf()),
-                     ('clf', FitParamT())])
+    pipe = Pipeline(
+        [("transf1", Transf()), ("transf2", Transf()), ("clf", FitParamT())],
+        memory="123",
+        verbose=True,
+    )
     pipe2 = pipe[:-1]
     assert isinstance(pipe2, Pipeline)
     assert pipe2.steps == pipe.steps[:-1]
     assert 2 == len(pipe2.named_steps)
+    assert pipe2.memory == pipe.memory
+    assert pipe2.verbose == pipe.verbose
```

Comment:
Can we more generally check for all parameters (use `get_params`) to ensure that this behaviour is maintained is we add more parameters to Pipeline? Thanks

Reply:
Well, I tried implementing that and it seems like it cannot work.  This is because `get_params` returns the parameters for the steps in addition to the direct pipeline parameters like `memory` and `verbose`.  Which means that 
1. I canot not test the equivalence between `pipe.get_params()` and `pipe2.get_params()` because the `"steps"` will be different
1. Even if we compare everything but `"steps"`, `pipe2.get_params()` will simply not have the parameters from the last step `"clf__whatever"` we remove in the tests by slicing it out.
1. For the same reason I cannot change the constructor call to something like `self.__class__(**self.get_params())`, with the `"steps"` key modified.

Altogether I'm leaning towards leaving the PR and the tests it as is.

Reply:
We can inspect `__init__` and skip check everything but the `steps` parameter:

```py
import inspect

def test_pipeline_slice():
	...

    paramters = inspect.signature(pipe.__init__).parameters
    for key, param in paramters.items():
        if key == 'steps':
            continue
        assert getattr(pipe2, key) == getattr(pipe, key)
```

Reply:
I think we should be asserting the equivalence of `get_params(deep=False)` except for steps.
```py
    pipe1_params = pipe.get_params(deep=False)
    del pipe1_params["steps"]
    pipe2_params = pipe.get_params(deep=False)
    del pipe2_params["steps"]
    assert pipe1_params == pipe2_params
```